### PR TITLE
chore(deps): Bump the production-dependencies group with 12 updates

### PR DESCRIPTION
**Pull request by ksimon1**
**What this PR does / why we need it**:
Replaces https://github.com/kubevirt/ssp-operator/pull/846
Fixes code after dependency update - there was change in behaviour of kubernetes-sigs/controller-runtime -
https://github.com/kubernetes-sigs/controller-runtime/pull/2633.
This commit updates test to copy TypeMeta in update CreateOrUpdate
function.

**Release note**:
```
None
```

### CODE REVIEW

**Review by 0xFelix — COMMENTED** [review]
/approve

File: go.mod [hunk 1]
```diff
@@ -23,18 +23,18 @@ require (
...
+	k8s.io/client-go v0.29.1
+	k8s.io/kube-aggregator v0.29.1
 	k8s.io/utils v0.0.0-20231127182322-b307cd553661
 	kubevirt.io/api v1.1.1
 	kubevirt.io/containerized-data-importer-api v1.58.0
 	kubevirt.io/controller-lifecycle-operator-sdk/api v0.2.4
 	kubevirt.io/qe-tools v0.1.8
 	kubevirt.io/ssp-operator/api v0.0.0
-	sigs.k8s.io/controller-runtime v0.16.3
+	sigs.k8s.io/controller-runtime v0.17.0
```
Comment (akrejcir) [inline go.mod:37]: This update requires golang version `1.21`: https://github.com/kubernetes-sigs/controller-runtime/blob/v0.17.0/go.mod

**Pull request merged**